pkg/gardenlet/controller/shoot/shoot/reconciler_delete.go

```diff
@@ -229,12 +232,17 @@ func (r *Reconciler) runDeleteShootFlow(ctx context.Context, o *operation.Operat
 		// controller-manager to be updateable due to provider config injection.
 		deployControlPlane = g.Add(flow.Task{
 			Name:         "Deploying Shoot control plane",
-			Fn:           flow.TaskFn(botanist.DeployControlPlane).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded),
+			Fn:           flow.TaskFn(botanist.DeployControlPlane).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded).SkipIf(o.Shoot.IsWorkerless),
 			Dependencies: flow.NewTaskIDs(initializeSecretsManagement, deployCloudProviderSecret, ensureShootClusterIdentity),
 		})
 		waitUntilControlPlaneReady = g.Add(flow.Task{
-			Name:         "Waiting until Shoot control plane has been reconciled",
-			Fn:           flow.TaskFn(botanist.Shoot.Components.Extensions.ControlPlane.Wait).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded),
+			Name: "Waiting until Shoot control plane has been reconciled",
+			Fn: flow.TaskFn(func(ctx context.Context) error {
+				if o.Shoot.IsWorkerless {
+					return nil
+				}
+				return botanist.Shoot.Components.Extensions.ControlPlane.Wait(ctx)
+			}).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded),
```

Comment:
```suggestion
			Fn:           flow.TaskFn(botanist.Shoot.Components.Extensions.ControlPlane.Wait).DoIf(cleanupShootResources && controlPlaneDeploymentNeeded).SkipIf(o.Shoot.IsWorkerless),
```
Why not the same way as in the previous section?
The same question for the other occurrences of 
```
 				if o.Shoot.IsWorkerless {
					return nil
				}
```
in the flow.

Reply:
`botanist.Shoot.Components.Extensions.ControlPlane` is not populated in workerless case. Hence we can't call Wait.

Reply:
Ah, I see 😅

Reply:
Isn't this a risk for creating nil pointer exceptions in gardenlet?

Reply:
This is to avoid nil pointer exceptions. We do similar checks for `skipReadiness` and `hibernation` too.